app/assets/javascripts/modules/base-target.js

```diff
@@ -0,0 +1,26 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function BaseTarget () { }
+
+  BaseTarget.prototype.start = function () {
+    if (this.pageIsInIframe()) {
+      this.setBaseTarget()
+    }
+  }
+
+  BaseTarget.prototype.setBaseTarget = function () {
+    var base = document.createElement('base')
```

Comment:
This will fix relative URLs but not absolute URLs, so if a content editor links to `http://www.example.com`, they will still get a preview issue. Would it be better to iterate over all `a` elements and set the `target` on them instead?

Reply:
Oh, darn.. I was trying to avoid doing that, but doesn't seem to cover all the links.

Reply:
I've just tested again now and it's working consistently regardless of links having absolute or relative paths — essentially it works for every link within the page.

Reply:
Ahh that's good. Thanks 👍 might be worth adding some relative, absolute and external links to the tests and validate that they all open in new tabs?

Reply:
Sounds like a good idea, but how to achieve that with Jasmine - as we don't have any real browser to test with? I know how to do it in Jest, but otherwise open to proposals.

Reply:
Easier said than done! I tried two approaches:

1) [spying on `window.open`](https://stackoverflow.com/questions/31286162/how-to-test-window-open-using-jasmine) - no longer seems to be supported by Jasmine (`toHaveBeenCalled is undefined`)
2) [monitoring the browser history length](https://www.sitepoint.com/community/t/find-out-when-page-is-opened-in-new-window-or-new-tab/8910/6) - if it's more than 1 then it opened in the current window - but this interferes with the running of the Jasmine test as we've navigated away from the Jasmine spec runner 🤦‍♂ 

Can't think of another way of testing it at the moment, other than manually, which I suppose will have to do!

Reply:
Tried spying on `window.open` too, but no luck. This adds to our set of cases where the limitations of Jasmine are blocking us from doing more robust testing. If it was part of an application we could rely on feature/integration tests, but otherwise I don't see an alternative either.